lldb/source/Plugins/SymbolFile/DWARF/AppleDWARFIndex.cpp

```diff
@@ -57,7 +57,9 @@ std::unique_ptr<AppleDWARFIndex> AppleDWARFIndex::Create(
     return std::make_unique<AppleDWARFIndex>(
```

Comment:
DataExtractor objects don't always have shared buffers. I am guessing in this case it does have them, but you might throw in a `lldbassert(...)` just in case to make sure this doesn't regress in the future.  Maybe something like:
```
lldbassert(apple_names.GetSharedDataBuffer());
lldbassert(apple_namespaces.GetSharedDataBuffer());
lldbassert(apple_types.GetSharedDataBuffer());
lldbassert(apple_objc.GetSharedDataBuffer());
```

Reply:
These asserts might not always pass? I don't know if it's guaranteed that if one of those sections is present, then the rest of them are as well.

Reply:
@clayborg what do you think about moving ownership of the data to inside the AppleAcceleratorTable class?

Reply:
Fine with me!

Reply:
> These asserts might not always pass? I don't know if it's guaranteed that if one of those sections is present, then the rest of them are as well.

You can check the size first and if non-zero then check the buffer:
```
lldbassert(apple_names.GetByteSize() == 0 || apple_names.GetSharedDataBuffer());
lldbassert(apple_namespaces.GetByteSize() == 0 || apple_namespaces.GetSharedDataBuffer());
lldbassert(apple_types.GetByteSize() == 0 || apple_types.GetSharedDataBuffer());
lldbassert(apple_objc.GetByteSize() == 0 || apple_objc.GetSharedDataBuffer());
```

Reply:
Without expressing an opinion, the [contributing guidelines](https://lldb.llvm.org/resources/contributing.html) say not to add new `lldbasserts`. This is currently being discussed on [Discourse](https://discourse.llvm.org/t/rfc-error-handling-in-release-builds-aka-can-i-use-lldbassert-or-not/74738/).

Reply:
@JDevlieghere , what do you suggest in this case? Plain assert or better not to put anything? I'd rather follow the guidelines

Reply:
This checks internal consistency so an `assert` is totally appropriate imho.

Reply:
Sure, I've just changed it to assert